ingester/ingester.go

```diff
@@ -73,19 +67,88 @@ var (
 	ErrDuplicateSampleForTimestamp = fmt.Errorf("sample with repeated timestamp but different value")
 )
 
+// Config for an Ingester.
+type Config struct {
+	ringConfig       ring.Config
+	userStatesConfig UserStatesConfig
+
+	// Config for the ingester lifecycle contol
+	ListenPort      *int
```

Comment:
Its shared with the server config - see https://github.com/weaveworks/cortex/blob/master/cmd/ingester/main.go#L33